src/main/java/me/zeroeightsix/fiber/tree/ConfigNode.java

```diff
@@ -17,6 +20,22 @@
      */
     String getName();
 
+    /**
+     * Returns the map describing the attributes of this node.
+     *
+     * <p> Attributes store metadata pertaining to the node itself, rather than its value.
+     * The returned map can be mutated by third parties to supplement the default node metadata.
+     * Examples of attributes include translation keys or rendering information.
+     *
+     * <p> As the returned data structure is shared by every attribute source,
+     * attributes should be grouped by namespace.
+     *
+     * @return this node's configurable attributes
+     */
+    Map<Identifier, ConfigAttribute<?>> getAttributes();
+
+    <A> ConfigAttribute<A> getOrCreateAttribute(Identifier id, @Nonnull A defaultValue);
+
```

Comment:
See comment below about subtyping. There is no reliable way to get the class object for a given type from an object compatible with that type.